### PR TITLE
support for vectors in macro defines (ie: "name:='a b c'")

### PR DESCRIPTION
**Pull request by outchy**
support for vectors in macro defines (ie: "name:='a b c'")

### CODE REVIEW

File: src/XacroParser.js [hunk 1]
```diff
@@ -214,7 +214,11 @@ export class XacroParser {
                 }
 
                 obj.name = name;
-                obj.def = def;
+                if (def.startsWith('\'') && def.endsWith('\'')) {
+                    obj.def = def.substr(1, def.length - 2);
```
Comment (gkjohnson) [inline src/XacroParser.js:218]: Can we use "substring" here, instead. "substr" is listed as legacy in the [mdn docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr).

File: src/XacroParser.js [hunk 2]
```diff
@@ -234,7 +238,7 @@ export class XacroParser {
             if (params) {
                 const inputs = params
                     .trim()
-                    .split(/\s+/g)
+                    .match(/[^\s']+(['][^']*['])?/g)
```
Comment (gkjohnson) [inline src/XacroParser.js:241]: Do the single quotes need to be in brackets `[']`?

**Review by gkjohnson — CHANGES_REQUESTED** [review]
Thanks for the contribution! A couple changes but overall it looks good. Would you be able to add a test or two for this feature, as well?

**Comment by gkjohnson** [comment]
I'm about to make a new release so I'll make some of the remaining changes. Thanks again for the fix and contribution!

**Comment by outchy** [comment]
Hello,

Apparently, everything is fine with your changes.

Thank you and sorry for the delay.

Cheers,

- Florent Ouchet

Le 24/04/2021 à 00:01, Garrett Johnson a écrit :

**Pull request merged**